feg/gateway/services/n7_n40_proxy/servicers/create_session.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/golang/glog"
+
+	n7_sbi "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControl"
+	"magma/feg/gateway/services/n7_n40_proxy/metrics"
+	"magma/feg/gateway/services/n7_n40_proxy/n7"
+	"magma/lte/cloud/go/protos"
+)
+
+// getSmPolicyRulesIfEnabled sends the request to PCF and get the response when N7 is enabled,
+// otherwise returns an empty SmPolicyDecision
+// Returns SmPolicyDecision rules and smPolicyUrl that uniquely identifies the policy session.
+func (srv *CentralSessionController) getSmPolicyRulesIfEnabled(
+	request *protos.CreateSessionRequest,
+) (*n7_sbi.SmPolicyDecision, string, error) {
+	if srv.cfg.N7Config.DisableN7 {
+		// Empty response when disabled
+		return &n7_sbi.SmPolicyDecision{
+			PccRules: &n7_sbi.SmPolicyDecision_PccRules{},
+		}, "", nil
+	}
+	// Convert and send the request to PCF
+	reqBody := n7.GetSmPolicyContextDataN7(request, srv.cfg.N7Config.Client.NotifyApiRoot)
+	reqCtx, cancel := context.WithTimeout(context.Background(), srv.cfg.RequestTimeout)
+	defer cancel()
+	resp, err := srv.policyClient.PostSmPoliciesWithResponse(reqCtx, *reqBody)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			metrics.N7Timeouts.Inc()
+		}
+		return nil, "", err
+	}
+	if resp.StatusCode() != http.StatusOK && resp.StatusCode() != http.StatusCreated {
+		err = fmt.Errorf("SmPolicyCreate request failure: status-code=%d", resp.StatusCode())
+		return nil, "", err
+	}
+	smPolicyUrl, found := resp.HTTPResponse.Header["Location"]
+	if !found {
+		err = fmt.Errorf("SmPolicyCreate request failure: Location header not found")
+		return nil, "", err
+	}
+	return resp.JSON201, smPolicyUrl[0], nil
+}
+
+func (srv *CentralSessionController) injectOmnipresentRules(
+	smPolicyDecision *n7_sbi.SmPolicyDecision,
+) error {
+	// No Base-Names returned in N7 response, however since these are statically configured,
+	// fetch all the omnipresent rules including the ones referred using the base-names. Will
+	// achieve the same behavior for both Gx and N7.
+	omnipresentRuleIDs, omnipresentBaseNames := srv.dbClient.GetOmnipresentRules()
+	if len(omnipresentRuleIDs) == 0 {
+		return nil
+	}
+	glog.V(2).Infof("Adding omnipresent rules %v omnipresent basenames %v", omnipresentRuleIDs, omnipresentBaseNames)
```

Comment:
please check the comment i made about this earlier

Reply:
I'm sorry I'm not able to identify the previous comment. Can you please point out which one it is?

Reply:
Per what you mentioned BaseNames do not exist on 5g. Why do we need them then?

And if you are injecting the definition of all the rules from the base name, can you point me where?

Reply:
Rules Ids of all BaseName groups are obtained in line no 71 (next line).

All static rules including the ones obtained from base name grouping is set in line no 74. 
`policy.PccRules.Set(ruleId, n7_sbi.PccRule{` 

The static rules that are set in the PccRules are converted to Static Rules proto in `model_conversion.go:getPccRules()` (#10298). `getpccRules()` is invoked when `GetCreateSessionResponseProto()` is invoked after `injectOmnipresentRules()`.

Reply:
Got it, i was missing `omnipresentBaseNames` are set on orc8r.